entc/gen/func.go

```diff
@@ -418,7 +418,24 @@ func dumpFields(v interface{}) string {
 		}
 		fv := rv.Field(i)
 		if !fv.IsZero() {
-			fields = append(fields, fmt.Sprintf("%s: %#v", f.Name, fv.Interface()))
+			var value string
+			if fv.Kind() == reflect.Ptr {
+				fv = reflect.Indirect(fv)
+				switch fv.Kind() {
+				case reflect.Bool:
+					value = fmt.Sprintf("boolPtr(%#v)", fv.Interface())
+				case reflect.String:
+					value = fmt.Sprintf("stringPtr(%#v)", fv.Interface())
+				case reflect.Int:
+					value = fmt.Sprintf("intPtr(%#v)", fv.Interface())
+				default:
+					value = fmt.Sprintf("%#v", fv.Interface())
+				}
+			} else {
+				value = fmt.Sprintf("%#v", fv.Interface())
+			}
+
+			fields = append(fields, fmt.Sprintf("%s: %s", f.Name, value))
 		}
```

Comment:
This won't work if we call this function from package `migrate`.  I think it will be simpler to just copy the `entsql` annotation in the [init function](https://github.com/facebook/ent/blob/master/entc/gen/template/migrate/schema.tmpl#L113), or we can do something like `F: &[]bool{true}[0]`, which is a bit ugly.

Reply:
I like `&[]bool{true}[0]` more than the helpers I guess :P

Reply:
changed to `[]...{}[0]`